react-native/react/login2/register/set-public-name/index.js

```diff
@@ -15,11 +15,14 @@ export default class SetPublicName extends Component {
     }
   }
 
-  submit () {
-    this.props.submit(this.state.deviceName)
+  onSubmit () {
+    this.props.onSubmit(this.state.deviceName)
   }
 
   render () {
+    const dupeName = this.props.existingDevices && this.props.existingDevices.indexOf(this.state.deviceName) !== -1
```

Comment:
i think there's an overall question of how we handle errors and who should do what when. i don't think we want to necessarily send calls to go on every text input to do validation. i agree about the dupe stuff, i'll make a ticket